src/search/search-index/add.js

```diff
@@ -180,3 +180,21 @@ async function performIndexing(indexDoc) {
         console.error(err)
     }
 }
+
+/**
+ * Adds a new meta index (bookmark or visit) entry.
+ * NOTE: Assumes the existence of indexed `pageId`.
+ *
+ * @param {'bookmark'|'visit'} type Type of meta event.
+ * @param {string} timestamp Timestamp of meta event.
+ * @param {string} pageId ID of page doc to associate with.
+ */
+export const addMetaConcurrent = (type, timestamp, pageId) =>
```

Comment:
I realised a problem with this: it will index the new visit, but it won't update the existing `pageId` page (which also needs a reference to the new visit as it acts as a reverse index). I will update this later and send you a code snippet (or feel free to do it yourself if you want - just needs to `page = await index.get(pageId)`, add the visit to `page.visits` then `index.put(page)`)